app/routes/questions/$question.tsx

```diff
@@ -199,55 +212,85 @@ function Contents({pageid, html, glossary}: {pageid: PageId; html: string; gloss
     return popup
   }
 
-  // The glossary items have to be injected somewhere, so this does it by manually wrapping any known
-  // definitions with spans. This is done from the longest to the shortest to make sure that sub strings
-  // of longer definitions don't override them.
-  const processedHTML = Object.keys(glossary)
-    .sort((a, b) => b.length - a.length)
-    .reduce(
-      (html, entry) =>
-        html.replace(
-          new RegExp(`(^|\\W)(${entry})($|\\W)`, 'gi'),
-          '$1<span class="glossary-entry">$2</span>$3'
-        ),
-      html
-    )
-
   useEffect(() => {
     const el = elementRef.current
     if (!el) return
 
-    el.querySelectorAll('.glossary-entry').forEach((e) => {
-      const entry = e.textContent && glossary[e?.textContent.toLowerCase().trim()]
-      if (
-        // It's possible for a glossary entry to contain another one (e.g. 'goodness' and 'good'), so
-        // if this entry is a subset of a bigger entry, remove it.
-        e.parentElement?.classList.contains('glossary-entry') ||
-        // Remove entries that are parts of external links, as that could get confusing
-        e.parentElement?.tagName == 'A' ||
-        // Remove entries that point to the current question
-        pageid == (entry as GlossaryEntry)?.pageid ||
-        !entry
-      ) {
-        e.outerHTML = e.textContent || ''
-      } else {
-        addPopup(
-          e as HTMLSpanElement,
-          `<div>${entry.contents}</div><br><a href="/?state=${entry.pageid}_">See more...</a>`
+    /*
+     * Replace all known glossary words in the given `textNode` with:
+     *  - a span to mark it as a glossary item
+     *  - an on hover popup with a short explaination of the glossary item
+     */
+    const insertGlossary = (textNode: Node) => {
```

Comment:
I very much dislike this function. There is a lot too much string -> DOM -> string -> DOM going on. But I couldn't think of a better way of inserting stuff. The main issue are cases like "good" and "goodness" which mess up most solutions